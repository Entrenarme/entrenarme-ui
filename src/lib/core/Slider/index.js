// @flow
import * as React from 'react';
import ReactSlider from 'react-slick';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import { faAngleRight, faAngleLeft } from '@fortawesome/free-solid-svg-icons';

import 'react-slick/lib/css/slick-theme.css';
import 'react-slick/lib/css/slick.css';

type Props = {
  children: React.Node,
};

type ArrowProps = {
  className?: string,
  style?: Object,
  onClick?: Function,
  icon: React.Node,
};

const Arrow = ({ className, style, onClick, icon }: ArrowProps) => (
  <div
    className={className}
    style={{
      ...style,
    }}
    onClick={onClick}
  >
    <FontAwesomeIcon icon={icon} />
  </div>
);

const Slider = ({ children, ...rest }: Props) => {
  const settings = {
    dots: true,
    slidesToShow: 3,
    slidesToScroll: 3,
    infinite: React.Children.count(children) >= 3 ? true : false,
    nextArrow: <Arrow icon={faAngleRight} />,
    prevArrow: <Arrow icon={faAngleLeft} />,
  };

  return <ReactSlider {...settings}>{children}</ReactSlider>;
};

Slider.defaultProps = {};

export default Slider;
