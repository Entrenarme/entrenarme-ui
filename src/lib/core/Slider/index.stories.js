// @flow
import * as React from 'react';
import { storiesOf } from '@storybook/react';
import styled from 'styled-components';

import Slider from './index';

const MainCard = styled.div`
  background-color: #fff;
  padding: 10px;
  border-radius: 4px;
  margin: 10px;
`;

const moreThan3 = [1, 2, 3, 4, 5];

const lessThan3 = [1, 2];

const ElementDecoration = storyFn => (
  <div style={{ backgroundColor: '#F1F1F1', height: '100vh' }}>{storyFn()}</div>
);

storiesOf('Slider', module)
  .addDecorator(ElementDecoration)
  .add('with more than 3 elements', () => (
    <Slider>
      {moreThan3.map(element => (
        <div>
          <MainCard>
            Number: {element}
            <br />
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
            fermentum magna eu velit semper hendrerit.
          </MainCard>
        </div>
      ))}
    </Slider>
  ));

storiesOf('Slider', module)
  .addDecorator(ElementDecoration)
  .add('with less than 3 elements', () => (
    <Slider>
      {lessThan3.map(element => (
        <div>
          <MainCard>
            Number: {element}
            <br />
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
            fermentum magna eu velit semper hendrerit.
          </MainCard>
        </div>
      ))}
    </Slider>
  ));
