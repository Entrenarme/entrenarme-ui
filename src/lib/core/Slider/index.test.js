import * as React from 'react';
import { render } from 'react-testing-library';
import 'react-testing-library/cleanup-after-each';
import styled from 'styled-components';

import Slider from './index';

const MainCard = styled.div`
  background-color: #fff;
  padding: 10px;
  border-radius: 4px;
  margin: 10px;
`;

const moreThan3 = [1, 2, 3, 4, 5];

test('renders "Slider" with 5 elements', () => {
  const { container } = render(
    <Slider itemsToShow={3}>
      {moreThan3.map(element => (
        <div key={element}>
          <MainCard>
            Number: {element}
            <br />
            Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec
            fermentum magna eu velit semper hendrerit.
          </MainCard>
        </div>
      ))}
    </Slider>,
  );

  expect(container.firstChild).toMatchSnapshot();
});
