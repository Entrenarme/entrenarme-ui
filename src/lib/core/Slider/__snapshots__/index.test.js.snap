// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders "Slider" with 5 elements 1`] = `
<div
  class="slick-initialized slick-slider sc-bdVaJa gqbKCE"
>
  <div
    class="slick-arrow slick-prev"
    style="display: block;"
  >
    <svg
      aria-hidden="true"
      data-prefix="fas"
      data-icon="angle-left"
      class="svg-inline--fa fa-angle-left fa-w-8 "
      role="img"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 256 512"
    >
      <path
        fill="currentColor"
        d="M31.7 239l136-136c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9L127.9 256l96.4 96.4c9.4 9.4 9.4 24.6 0 33.9L201.7 409c-9.4 9.4-24.6 9.4-33.9 0l-136-136c-9.5-9.4-9.5-24.6-.1-34z"
      />
    </svg>
  </div>
  <div
    class="slick-list"
  >
    <div
      class="slick-track"
      style="opacity: 1;"
    >
      <div
        data-index="-3"
        class="slick-slide slick-cloned"
        style="width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          3
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="-2"
        class="slick-slide slick-cloned"
        style="width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          4
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="-1"
        class="slick-slide slick-cloned"
        style="width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          5
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="0"
        class="slick-slide slick-active"
        tabindex="-1"
        style="outline: none; width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          1
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="1"
        class="slick-slide slick-active"
        tabindex="-1"
        style="outline: none; width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          2
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="2"
        class="slick-slide slick-active"
        tabindex="-1"
        style="outline: none; width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          3
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="3"
        class="slick-slide"
        tabindex="-1"
        style="outline: none; width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          4
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="4"
        class="slick-slide"
        tabindex="-1"
        style="outline: none; width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          5
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="5"
        class="slick-slide slick-cloned"
        style="width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          1
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="6"
        class="slick-slide slick-cloned"
        style="width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          2
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
      <div
        data-index="7"
        class="slick-slide slick-cloned"
        style="width: 0px;"
      >
        <div
          class="sc-bwzfXH iOpBWD"
        >
          Number: 
          3
          <br />
          Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec fermentum magna eu velit semper hendrerit.
        </div>
      </div>
    </div>
  </div>
  <div
    class="slick-arrow slick-next"
    style="display: block;"
  >
    <svg
      aria-hidden="true"
      data-prefix="fas"
      data-icon="angle-right"
      class="svg-inline--fa fa-angle-right fa-w-8 "
      role="img"
      xmlns="http://www.w3.org/2000/svg"
      viewBox="0 0 256 512"
    >
      <path
        fill="currentColor"
        d="M224.3 273l-136 136c-9.4 9.4-24.6 9.4-33.9 0l-22.6-22.6c-9.4-9.4-9.4-24.6 0-33.9l96.4-96.4-96.4-96.4c-9.4-9.4-9.4-24.6 0-33.9L54.3 103c9.4-9.4 24.6-9.4 33.9 0l136 136c9.5 9.4 9.5 24.6.1 34z"
      />
    </svg>
  </div>
  <ul
    class="slick-dots"
    style="display: block;"
  >
    <li
      class="slick-active"
    >
      <button>
        1
      </button>
    </li>
    <li
      class=""
    >
      <button>
        2
      </button>
    </li>
  </ul>
</div>
`;
